CMakeLists.txt

```diff
@@ -12,6 +12,11 @@ execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_SOURCE_DIR}/l
 set(LIBRARY_OUTPUT_PATH ${CMAKE_SOURCE_DIR}/lib)
 
 set(CMAKE_OSX_ARCHITECTURES x86_64)
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
+if(POLICY CMP0063)
```

Comment:
This is an unfortunate consequence of using cmake.  The only reason this is here is to make it not spam warnings when building with newer versions.

I could add a generic comment explaining that, but I won't explain what each policy does individually, that would be too much work and is already easily searchable on the internet for the curious.